File: PreUtil/src/output/java/easy/KeyboardRow.java
/**https://leetcode.com/problems/keyboard-row/ */
//Given an array of strings words, return the words that can be typed using letters of the alphabet on only one row of American keyboard like the image below.
//In the American keyboard:
//the first row consists of the characters "qwertyuiop",
//the second row consists of the characters "asdfghjkl", and
//the third row consists of the characters "zxcvbnm".


class KeyboardRow {
    public String[] findWords(String[] words) {
        String[] rows = {"qwertyuiop", "asdfghjkl", "zxcvbnm"};
        List<String> ans = new ArrayList<>();
        for(String word : words){
            int row = 0;
            for(int i = 0; i < rows.length; i++){
                if(rows[i].indexOf(Character.toLowerCase(word.charAt(0))) != -1){
                    row = i;
                    break;
                }
            }
            boolean flag = true;
            for(int i = 1; i < word.length(); i++){
                if(rows[row].indexOf(Character.toLowerCase(word.charAt(i))) == -1){
                    flag = false;
                    break;
                }
            }
            if(flag){
                ans.add(word);
            }
        }
        return ans.toArray(new String[ans.size()]);
        
        
    }
}

    

File: PreUtil/src/output/python/easy/KeyboardRow.py
#https://leetcode.com/problems/keyboard-row/
#Given an array of strings words, return the words that can be typed using letters of the alphabet on only one row of American keyboard like the image below.
#In the American keyboard:
#the first row consists of the characters "qwertyuiop",
#the second row consists of the characters "asdfghjkl", and
#the third row consists of the characters "zxcvbnm".


class Solution(object):
    def findWords(self, words):
        """
        :type words: List[str]
        :rtype: List[str]
        """      
        res=[]
        for i in range(len(words)):
            if words[i][0].lower() in "qwertyuiop":
                for j in range(len(words[i])):
                    if words[i][j].lower() not in "qwertyuiop":
                        break
                else:
                    res.append(words[i])
            elif words[i][0].lower() in "asdfghjkl":
                for j in range(len(words[i])):
                    if words[i][j].lower() not in "asdfghjkl":
                        break
                else:
                    res.append(words[i])
            elif words[i][0].lower() in "zxcvbnm":
                for j in range(len(words[i])):
                    if words[i][j].lower() not in "zxcvbnm":
                        break
                else:
                    res.append(words[i])
        return res
       

    